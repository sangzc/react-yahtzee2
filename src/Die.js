import React, { Component } from "react";
import "./Die.css";

class Die extends Component {
  render() {
    return (
      <button
        className={this.props.locked ? "Die Die-locked" : "Die"}
        onClick={() => this.props.handleClick(this.props.idx)}>
        {this.props.val}
      </button>
    );
  }
}

export default Die;
