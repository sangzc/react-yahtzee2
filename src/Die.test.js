import React from 'react';
import { mount } from "enzyme";
import Die from "./Die";
import toJson from "enzyme-to-json"


it('renders without crashing', () => {
    mount(<Die value="1"/>);
});

it('locks die', () => {
    let wrapper = mount(<Die locked value="1"/>);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it('keeps die unlocked', () => {
    let wrapper = mount(<Die value="1"/>);
    let serialized = toJson(wrapper);
    expect(serialized).toMatchSnapshot();
});

it('does clicking work test', () => {
    let clickFn = jest.fn();
    let wrapper = mount(<Die handleToggleLocked={clickFn} value="1"/>);
    wrapper.simulate("click");
    expect(clickFn).toHaveBeenCalled();
});




