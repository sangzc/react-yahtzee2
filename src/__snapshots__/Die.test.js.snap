// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`keeps die unlocked 1`] = `
<Die
  value="1"
>
  <button
    className="Die"
    onClick={[Function]}
  />
</Die>
`;

exports[`locks die 1`] = `
<Die
  locked={true}
  value="1"
>
  <button
    className="Die Die-locked"
    onClick={[Function]}
  />
</Die>
`;
